File: app/src/main/java/mk/webfactory/template/feature/launcher/ui/LauncherActivity.java
/*
 * MIT License
 *
 * Copyright (c) 2020 Web Factory LLC
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package mk.webfactory.template.feature.launcher.ui;

import android.content.Intent;
import android.os.Bundle;
import javax.inject.Inject;
import mk.webfactory.template.R;
import mk.webfactory.template.feature.common.ui.BaseActivity;
import mk.webfactory.template.feature.home.ui.HomeActivity;
import mk.webfactory.template.user.UserManager;

public class LauncherActivity extends BaseActivity {

  @Inject
  UserManager userManager;

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_container);

    startActivity(new Intent(LauncherActivity.this, HomeActivity.class));
    if (userManager.isLoggedIn()) {
      //startActivity(new Intent(LauncherActivity.this, HomeActivity.class));
      //finish();
    }
  }
}

File: app/src/main/java/mk/webfactory/template/feature/launcher/ui/LauncherActivity.kt
package mk.webfactory.template.feature.launcher.ui

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import dagger.hilt.android.AndroidEntryPoint
import mk.webfactory.template.R
import mk.webfactory.template.feature.home.ui.HomeActivity
import mk.webfactory.template.feature.login.ui.LoginActivity
import mk.webfactory.template.model.user.UserSession
import mk.webfactory.template.user.UserManager
import javax.inject.Inject

@AndroidEntryPoint
class LauncherActivity : AppCompatActivity() {

    //todo use navigation component

    @Inject
    lateinit var userManager: UserManager<UserSession>

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_container)
        if (userManager.isLoggedIn()) {
            HomeActivity.startActivity(this)
        } else {
            LoginActivity.startActivityNewTask(this)
        }
        finish()
    }
}